elb/src/test/java/org/jclouds/elb/features/SubnetApiLiveTest.java

```diff
@@ -16,13 +16,145 @@
  */
 package org.jclouds.elb.features;
 
+import static org.assertj.core.util.Preconditions.checkNotNull;
+import static org.jclouds.reflect.Reflection2.typeToken;
+
+import java.util.Iterator;
+import java.util.Random;
+import java.util.Set;
+
+import org.jclouds.ContextBuilder;
+import org.jclouds.ec2.EC2Api;
+import org.jclouds.ec2.compute.EC2ComputeServiceContext;
+import org.jclouds.ec2.domain.Subnet;
+import org.jclouds.elb.domain.Listener;
+import org.jclouds.elb.domain.LoadBalancer;
+import org.jclouds.elb.domain.Protocol;
 import org.jclouds.elb.internal.BaseELBApiLiveTest;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
 @Test(groups = "live", testName = "SubnetApiLiveTest")
 public class SubnetApiLiveTest extends BaseELBApiLiveTest {
+   private String loadBalancerName;
+   private EC2Api ec2Api;
+
+
+   @BeforeMethod
+   private void setUp() {
+      super.setup();
 
-   protected SubnetApi api() {
+      final EC2ComputeServiceContext ec2ComputeServiceContext = ContextBuilder.newBuilder("aws-ec2")
+              .credentials(identity, credential)
+              .buildView(typeToken(EC2ComputeServiceContext.class));
+
+     ec2Api = ec2ComputeServiceContext.unwrapApi(EC2Api.class);
+
+   }
+
+   private void createTestLoadBalancer (){
+      api.getLoadBalancerApi().createListeningInAvailabilityZones(loadBalancerName, Listener.builder()
+              .protocol(Protocol.HTTP)
+              .port(80)
+              .instanceProtocol(Protocol.HTTP)
+              .instancePort(80)
+              .build(), ImmutableList.of("us-east-1b"));
+   }
+   @AfterClass
```

Comment:
Better use `@AfterClass(alwaysRun = true)`, to make sure the cleanup happens even if tests fail. This way we do a best effort to avoid leaking resources generated during the tests.
Apply this to the other created live test classes that can benefit from this pattern.